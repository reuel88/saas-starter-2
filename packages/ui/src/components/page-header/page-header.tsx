import { cn } from "@repo/utilities";
import { FC, ReactNode } from "react";
import { LucideIcon } from "lucide-react";

type PageHeaderProps = {
  children?: ReactNode;
  title: string;
  description: string;
  icon: LucideIcon;
  iconColor?: string;
  bgColor?: string;
};

export const PageHeader: FC<PageHeaderProps> = ({
  children,
  title,
  description,
  icon: Icon,
  iconColor,
  bgColor,
}) => {
  return (
    <header className="my-8 flex items-stretch gap-x-6 px-4 lg:px-8">
      <div className="flex items-center gap-x-3">
        <div className="h-full flex-1">
          <div className={cn("w-fit rounded-md p-2", bgColor)}>
            <Icon className={cn("h-10 w-10", iconColor)} />
          </div>
        </div>
        <div>
          <h2 className="text-3xl font-bold">{title}</h2>
          <p className="max-w-[50ch] text-sm text-muted-foreground">
            {description}
          </p>
        </div>
      </div>

      {children && <div>{children}</div>}
    </header>
  );
};
