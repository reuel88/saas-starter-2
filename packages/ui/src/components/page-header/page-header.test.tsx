import { render, screen } from "@testing-library/react";
import { PageHeader } from "./page-header";
import { LucideStar } from "lucide-react";

describe("Container", () => {
  it("should render a header with a title and description", () => {
    // Arrange
    const title = "Test Title";
    const description = "Test Description";

    // Act
    const { baseElement } = render(
      <PageHeader title={title} description={description} icon={LucideStar} />,
    );

    // Assert
    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(description)).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
