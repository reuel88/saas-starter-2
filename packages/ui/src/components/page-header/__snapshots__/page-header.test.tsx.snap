// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > should render a header with a title and description 1`] = `
<body>
  <div>
    <header
      class="my-8 flex items-stretch gap-x-6 px-4 lg:px-8"
    >
      <div
        class="flex items-center gap-x-3"
      >
        <div
          class="h-full flex-1"
        >
          <div
            class="w-fit rounded-md p-2"
          >
            <svg
              class="lucide lucide-star h-10 w-10"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
          </div>
        </div>
        <div>
          <h2
            class="text-3xl font-bold"
          >
            Test Title
          </h2>
          <p
            class="max-w-[50ch] text-sm text-muted-foreground"
          >
            Test Description
          </p>
        </div>
      </div>
    </header>
  </div>
</body>
`;
